PokeAlarm/Cache/Cache.py

```diff
@@ -97,6 +98,12 @@ def day_or_night_id(self, s2_cell_id, day_or_night_id=Unknown.REGULAR):
             self._day_or_night_id[s2_cell_id] = day_or_night_id
         return self._day_or_night_id.get(s2_cell_id, Unknown.REGULAR)
 
+    def quest_reward(self, stop_id, reward=None):
```

Comment:
the uniqueness of quests is not only the stop_id + reward. There's also the quest/task itself.

the Cleanup method also needs some way to flush out all old rewards at midnight, or else pokestops that happen to have the same task/reward two days in a row will never get reported.